src/Control/Email/Email.php

```diff
@@ -31,738 +18,641 @@ class Email extends ViewableData
+     * @param string|array|null $to
      * @param string|null $subject
      * @param string|null $body
-     * @param string|null $bounceHandlerURL
-     * @param string|null $cc
-     * @param string|null $bcc
+     * @param string|array|null $cc
+     * @param string|array|null $bcc
+     * @param string|null $returnPath
      */
     public function __construct(
         $from = null,
         $to = null,
         $subject = null,
         $body = null,
-        $bounceHandlerURL = null,
         $cc = null,
-        $bcc = null
+        $bcc = null,
+        $returnPath = null
     ) {
-
-        if ($from !== null) {
-            $this->from = $from;
+        if ($from) {
+            $this->setFrom($from);
         }
-        if ($to !== null) {
-            $this->to = $to;
+        if ($to) {
+            $this->setTo($to);
         }
-        if ($subject !== null) {
-            $this->subject = $subject;
+        if ($subject) {
+            $this->setSubject($subject);
         }
-        if ($body !== null) {
-            $this->body = $body;
+        if ($body) {
+            $this->setBody($body);
         }
-        if ($cc !== null) {
-            $this->cc = $cc;
+        if ($cc) {
+            $this->setCC($cc);
         }
-        if ($bcc !== null) {
-            $this->bcc = $bcc;
+        if ($bcc) {
+            $this->setBCC($bcc);
         }
-
-        if ($bounceHandlerURL !== null) {
-            Deprecation::notice('4.0', 'Use "emailbouncehandler" module');
+        if ($returnPath) {
+            $this->setReturnPath($returnPath);
         }
 
         parent::__construct();
     }
 
     /**
-     * Get the mailer.
-     *
-     * @return Mailer
+     * @return Swift_Message
      */
-    public static function mailer()
+    public function getSwiftMessage()
     {
-        return Injector::inst()->get('SilverStripe\\Control\\Email\\Mailer');
+        if (!$this->swiftMessage) {
+            $this->setSwiftMessage(new Swift_Message());
+        }
+
+        return $this->swiftMessage;
     }
 
     /**
-     * Attach a file based on provided raw data.
+     * @param Swift_Message $swiftMessage
      *
-     * @param string $data The raw file data (not encoded).
-     * @param string $attachedFilename Name of the file that should appear once it's sent as a separate attachment.
-     * @param string|null $mimeType MIME type to use when attaching file. If not provided, will attempt to infer via HTTP::get_mime_type().
      * @return $this
      */
-    public function attachFileFromString($data, $attachedFilename, $mimeType = null)
+    public function setSwiftMessage($swiftMessage)
     {
-        $this->attachments[] = array(
-            'contents' => $data,
-            'filename' => $attachedFilename,
-            'mimetype' => $mimeType,
-        );
+        $swiftMessage->setDate(DBDatetime::now()->Format('U'));
+        if (!$swiftMessage->getFrom() && $defaultFrom = $this->config()->admin_email) {
```

Comment:
Operator precedence; Should the assignment `=` be surrounded with braces, since it's lower than the `&&`?